inc/fields/class-shortcode-ui-field-post-select.php

```diff
@@ -145,7 +145,7 @@ public function action_wp_ajax_shortcode_ui_post_field() {
 
 		$query                  = new WP_Query( $query_args );
 		$post_types             = $query->get( 'post_type' );
-		$is_multiple_post_types = count( $post_types ) > 1 || 'any' === $post_types;
+		$is_multiple_post_types = is_array( $post_types ) && count( $post_types ) > 1 || 'any' === $post_types;
```

Comment:
I haven't tested this, but it looks like this would cause an order-of-precedence issue where the last part of the conditional would never be reached because "any" is not an array.

Can we add some grouping here so that it's easier to read and doesn't change the meaning of the conditional?

```suggestion
		$is_multiple_post_types = ( is_array( $post_types ) && count( $post_types ) > 1 ) || 'any' === $post_types;
```